src/watchers.ts

```diff
@@ -0,0 +1,120 @@
+import * as core from '@actions/core';
+import axios from 'axios';
+
+class JiraWatcherManager {
+  private jiraIssueUrl: string;
+  private botToken: string;
+  private issueKey: string;
+  private autoWatchAccounts: string[];
+
+  constructor(jiraIssueUrl: string, botToken: string) {
+    const autoWatchAccounts = core.getInput('autoWatchAccounts');
```

Comment:
I don't have strong reasoning for this, but I liked having hall of the "inputs" to the reconcile-issue living at the very top of `reconcile-issue.ts` so it doesn't get lost.

Would it be possible to:
1. Accept the input in `reconcile-issue::reconcileIssue()`
2. Pass the watchers to Jira constructor
3. On create (and update) issue, the watchers are passed in when creating new JiraWatcherManagers.

Reply:
Yeah this is dependency injection and as we learned with the broker it can get really ugly when you plumb a dependency through as constructor arguments, especially as they get progressively more nested and pervasive (the "logger") issue.

I think your suggestion is a good one here though and I will update.

Reply:
after chatting about this one, I think it's a nack on this one (rightfully so). Marking as resolved.